FluentAssertions.Core/Types/TypeAssertions.cs

```diff
@@ -198,6 +200,366 @@ public AndConstraint<TypeAssertions> BeDecoratedWith<TAttribute>(string because
         }
 
         /// <summary>
+        /// Asserts that the current <see cref="Type"/> implements Interface <paramref name="interfaceType"/>.
+        /// </summary>
+        /// <param name="interfaceType">The interface that should be implemented.</param>
+        /// <param name="because">A formatted phrase as is supported by <see cref="M:System.String.Format(System.String,System.Object[])"/> explaining why the assertion
+        ///             is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.</param>
+        /// <param name="reasonArgs">Zero or more objects to format using the placeholders in <see cref="!:because"/>.</param>
+        public AndConstraint<TypeAssertions> Implement(Type interfaceType, string because = "", params object[] reasonArgs)
+        {
+            if (!interfaceType.IsInterface)
```

Comment:
The abstract class scenario is handled by BeDerivedFrom. According to MSDN, types derive from types and implement interfaces. There is the special case of a purely abstract base class, but I still like the Derive verb since it implies that we're bound by single inheritance, which is true even in the special case.